### DISCUSSION TITLE
scrypted LXC not starting

### DISCUSSION THREAD

**Discussion by kobius77** [tteck/Proxmox#2872] [Q&A]
just found out about scrypted and thought i need to try it out - but it doesn't start up.

when started at the end of the script i got this:

`TASK ERROR: Device /dev/dri/renderD128 does not exist`

and trying to start again manually i get:

`TASK ERROR: Device /dev/dri/card1 does not exist`

(from then on it's one after the other)

granted, host is a rather weak Athlon 240GE with onboard graphics, but according to scrypted docs it seems that it has rather modest requirements. what am i missing?

thanks for any input.

**Comment by tteck** [comment]
What's the output of `ls -l /dev/dri` when executed within the Proxmox shell?

**Reply by kobius77** [reply]
No such file or directory

**Reply by tteck** [reply]
Until you figure out why that is (No such file or directory), anything using Hardware Acceleration will show those errors.
 
EXAMPLE:
```
root@server:~# ls -l /dev/dri
total 0
drw-rw---- 2 root root         80 Apr 24 09:18 by-path
crw-rw---- 1 root video  226,   1 Apr 24 09:18 card0
crw-rw---- 1 root render 226, 128 Apr 24 09:18 renderD128
```
I guess I could make Hardware Acceleration a selectable option.

**Comment by tteck** [comment]
Did you create the LXC using Default settings?

**Reply by kobius77** [reply]
just copy/pasted the bash command with the url - as usual ;)

**Comment by marcoue** [comment]
I came across the same issue. My only solution was to install the script on another Proxmox node and migrate it to the main one. I was then able to run it.

**Reply by ProtocolPro** [reply]
I will definitely have to try this option

**Comment by ProtocolPro** [comment]
I'm running into the same issue

**Reply by chenks** [reply]
What's the output of `ls -l /dev/dri` when executed within the Proxmox shell?

**Reply by ProtocolPro** [reply]
root@apex:~# ls -l /dev/dri
total 0
drwxr-xr-x 2 root root         80 Jul 19 07:41 by-path
crw-rw---- 1 root video  226,   0 Jul 19 07:41 card0
crw-rw---- 1 root render 226, 128 Jul 19 07:41 renderD128

**Reply by chenks** [reply]
You have card0 as opposed to card1, so make sure your devices match in the lxc settings

**Reply by ProtocolPro** [reply]
by change how do i change that? Only thing I did was reboot the proxmox environment then this happened

**Reply by chenks** [reply]
Yep, rebooting has caused your devices to switch 

2 options

1) manually adjust the lxc device settings to match what they now are (as you listed above)

2) reboot proxmox again, they might switch back again on their own

**Reply by ProtocolPro** [reply]
thank you!  option 1 worked for me

**Reply by chenks** [reply]
Edit the configured devices in the lxc settings

**Comment by tomduca** [comment]
I'm having the same/similar problem.
Immediately after installation it works ok. 
But after the first reboot I won't start anymore, showing following message:
`TASK ERROR: Device /dev/dri/renderD128 does not exist`

**In Proxmox host:**
root@pve1:~# lspci -v -s 00:02.0
00:02.0 VGA compatible controller: Intel Corporation CometLake-S GT2 [UHD Graphics 630] (rev 05) (prog-if 00 [VGA controller])
        DeviceName: Onboard - Video
        Subsystem: Dell CometLake-S GT2 [UHD Graphics 630]
        Flags: fast devsel, IRQ 16, IOMMU group 0
        Memory at 90000000 (64-bit, non-prefetchable) [size=16M]
        Memory at 80000000 (64-bit, prefetchable) [size=256M]
        I/O ports at 3000 [size=64]
        Expansion ROM at 000c0000 [virtual] [disabled] [size=128K]
        Capabilities: [40] Vendor Specific Information: Len=0c <?>
        Capabilities: [70] Express Root Complex Integrated Endpoint, MSI 00
        Capabilities: [ac] MSI: Enable- Count=1/1 Maskable- 64bit-
        Capabilities: [d0] Power Management version 2
        Capabilities: [100] Process Address Space ID (PASID)
        Capabilities: [200] Address Translation Service (ATS)
        Capabilities: [300] Page Request Interface (PRI)
        Kernel driver in use: **vfio-pci**
        Kernel modules: i915


**This is my lxc config:**

arch: amd64
cores: 4
dev0: /dev/dri/card1,gid=44
dev1: /dev/dri/renderD128,gid=104
features: keyctl=1,nesting=1
hostname: plexlxc
memory: 2048

I tried to change card1 to card0 but it says card0 does not exist.
I'm suspicious about Kernel driver in use. Should it be **i915** instead of current **vfio-pci**?
I believe before deploying the LXC it was **i915** in use.
If that's it, how do I do to change and force it to stay?

Thank you!

**Reply by tomduca** [reply]
(Didn't) Fixed it blacklisting vfio-pci in host and setting early bind to i915 in /etc/modprobe.d:

options i915 ids=8086:9bc5
softdep vfio-pci pre: i915

With this host always use i915 for it and LXC can always "see" it.

**Reply by tomduca** [reply]
Hello. Coming back again. It didn't work.
After almost a day, the Kernel driver in use changed back to vfio-cip, without rebooting the lxc neither the Proxmox host.
And obviously at the moment that is changed, the LXC goes down and it can't be started until Kernel drive in use is set back to i915. Which I don't know how to do it permanently.
Any help will be appreciated.

_root@pve1:~# lspci -v -s 00:02.0
00:02.0 VGA compatible controller: Intel Corporation CometLake-S GT2 [UHD Graphics 630] (rev 05) (prog-if 00 [VGA controller])
        DeviceName: Onboard - Video
        Subsystem: Dell CometLake-S GT2 [UHD Graphics 630]
        Flags: fast devsel, IRQ 16, IOMMU group 0
        Memory at 90000000 (64-bit, non-prefetchable) [size=16M]
        Memory at 80000000 (64-bit, prefetchable) [size=256M]
        I/O ports at 3000 [size=64]
        Expansion ROM at 000c0000 [virtual] [disabled] [size=128K]
        Capabilities: [40] Vendor Specific Information: Len=0c <?>
        Capabilities: [70] Express Root Complex Integrated Endpoint, MSI 00
        Capabilities: [ac] MSI: Enable- Count=1/1 Maskable- 64bit-
        Capabilities: [d0] Power Management version 2
        Capabilities: [100] Process Address Space ID (PASID)
        Capabilities: [200] Address Translation Service (ATS)
        Capabilities: [300] Page Request Interface (PRI)
        **Kernel driver in use: vfio-pci**
        Kernel modules: i915_

**Reply by tomduca** [reply]
It ended up being that the video device was being passed through to a VM. Once I removed the video device from that VM, I got rid of vfio-pci for good.
Sorry for the spam.
